t a/classes/AutoTest.js b/classes/AutoTest.js
index ca5517a..66781a4 100644
--- a/classes/AutoTest.js
+++ b/classes/AutoTest.js
@@ -34,17 +34,7 @@ class AutoTest {
   async gptFillQuery(text) {
     try {
       const page = await this.getPage('chatgpt.com');
-
-      // 点击输入框并输入文本
-      await page.click('#prompt-textarea');
-      await page.keyboard.type('  ');
-      // 传递 text 作为参数给 page.evaluate
-      await page.evaluate((text) => {
-        const inputEl = document.querySelector('#prompt-textarea');
-        if (inputEl) {
-          inputEl.value = text;
-        }
-      }, text);
+      await page.fill('#prompt-textarea', text)
       await page.click('[data-testid="send-button"]');
       
       // 等待 stop-button 可见
@@ -78,7 +68,8 @@ if(module === require.main){
   (async () => {
     const autoTest = new AutoTest();
     await autoTest.initialize();
-    const unswer = await autoTest.gptFillQuery('随机生成一篇200字的散文')
+    const text = '今天星期几'
+    const unswer = await autoTest.gptFillQuery(text)
     console.log(unswer);
     await autoTest.closeBrowser();
   })();
diff --git a/classes/Util.js b/classes/Util.js
index 749f355..1c230d5 100644
--- a/classes/Util.js
+++ b/classes/Util.js
@@ -36,6 +36,18 @@
         })
       })
     },
+
+    async gptAsk(text){
+      let res = await this.request({
+        url: `http://localhost:9000/gptFillQuery`,
+        method: 'post',
+        data: {
+          text
+        }
+      })
+      return res.data
+    }
+
   };
 
   return Util;
diff --git a/tampermonkeys/chatgpt/ChatgptApp.js b/tampermonkeys/chatgpt/ChatgptApp.js
index ded9b48..4efed37 100644
--- a/tampermonkeys/chatgpt/ChatgptApp.js
+++ b/tampermonkeys/chatgpt/ChatgptApp.js
@@ -9,6 +9,7 @@ const ChatgptApp = {
   template,
   data: {
     apiBaseUrl: 'https://node-gpt-h1b3.onrender.com',
+    // apiBaseUrl: 'https://8080-cs-239467590834-default.cs-asia-southeast1-ajrg.cloudshell.dev',
     novels: [],
   },
   mounted(){  
@@ -53,7 +54,12 @@ const ChatgptApp = {
         text: novel.cnCont
       }
     })
-    console.log(res)
+
+    let cnSection = `### ${novel.cnTitle} \n\n${novel.cnCont}`
+    let enSection = await Util.gptAsk(cnSection)
+    console.log(enSection)
+
+    
   }
 }
 

